macros/generic_tests/at_least_one.sql

```diff
@@ -4,38 +4,20 @@
 
 {% macro default__test_at_least_one(model, column_name, group_by_columns) %}
 
-{% set pruned_cols = [column_name] %}
-
 {% if group_by_columns|length() > 0 %}
-
   {% set select_gb_cols = group_by_columns|join(' ,') + ', ' %}
   {% set groupby_gb_cols = 'group by ' + group_by_columns|join(',') %}
-  {% set pruned_cols = group_by_columns %}
-
-  {% if column_name not in pruned_cols %}
-    {% do pruned_cols.append(column_name) %}
-  {% endif %}
-
 {% endif %}
 
-{% set select_pruned_cols = pruned_cols|join(' ,') %}
-
 select *
 from (
-    with pruned_rows as (
-      select
-        {{ select_pruned_cols }}
-      from {{ model }}
-      where {{ column_name }} is not null
-      limit 1
-    )
     select
         {# In TSQL, subquery aggregate columns need aliases #}
         {# thus: a filler col name, 'filler_column' #}
       {{select_gb_cols}}
       count({{ column_name }}) as filler_column
 
-    from pruned_rows
+    from {{ model }}
```

Comment:
Is there a way to add integration tests to validate that the test fails appropriately?

Reply:
Yes, adding a test would be the key thing here 🔑

[Here](https://github.com/dbt-labs/dbt-utils/blob/4da44cb930df07921e9c4335dba2e30f9e0802d4/integration_tests/models/generic_tests/schema.yml#L151-L152) is an example of using  `error_if` in a clever way.

### Explanation

By default, dbt treats:
- `0` row as `PASS`
- `>0` rows as `FAIL`

Overriding the config to `error_if: "<1"` will invert that. To the best of my knowledge, this feature was discovered rather than designed, and it also requires adding `warn_if: "<0"` to work properly.

Reply:
Sweet! Thanks, Doug. I've given this a go in https://github.com/dbt-labs/dbt-utils/pull/922/commits/1fe5fcfc28475f1addb1b038db8a529b71cfe30f.